### PR TITLE
[ME] UI components for attached resources

### PR DESCRIPTION
**Pull request by jahow**
### Description

This PR introduces two UI components that are necessary for the Attached Resources field:
- file input (either URL or upload)
- URL input

This also modifies the image input component to match the file input one and use the same components (URL and text inputs).

This PR introduces a global CSS class `gn-ui-text-input` which is used for all kinds of text inputs: simple text, textarea, url. this way, all these components show up in an homogenized way. CSS variables are available to customize the class if needed: `--gn-ui-text-input-rounded` and `--gn-ui-text-input-padding`.
Note: this is not used yet for the autocomplete field

**Potential breaking change**: the `hint` parameter of text-input and rich-input was changed to `placeholder` for consistency ("hint" is used in the editor form when talking about tooltips and help texts; placeholder is a more common term).

See mockups here: https://www.figma.com/design/7rQdQNSKqK45FM5MDrzt0B/GN4---Editor?node-id=3314-26706&t=LMryBb5l8KnnCzCL-0

### Architectural changes

Made `text-input` and `rich-input` standalone components; this had impacts in map-viewer components

### Screenshots

image

image

### Quality Assurance Checklist

- [x] Commit history is devoid of any _merge commits_ and readable to facilitate reviews
- [x] If **new logic** ⚙️ is introduced: unit tests were added
- [ ] If **new user stories** 🤏 are introduced: E2E tests were added
- [x] If **new UI components** 🕹️ are introduced: corresponding stories in Storybook were created
- [ ] If **breaking changes** 🪚 are introduced: add the `breaking change` label
- [ ] If **bugs** 🐞 are fixed: add the `backport <release branch>` label
- [ ] The [documentation website](docs) 📚 has received the love it deserves

---

**This work is sponsored by [Organization ABC](xx)**.

### CODE REVIEW

File: libs/ui/inputs/src/lib/url-input/url-input.component.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  standalone: true,
+  imports: [CommonModule, ButtonComponent, MatIconModule],
+})
+export class UrlInputComponent {
+  @Input() value = ''
+  @Input() extraClass = ''
+  @Input() placeholder = 'https://'
+  @Input() disabled: boolean
+  rawChange = new Subject<string>()
+  @Output() valueChange = this.rawChange.pipe(filter((v) => !!v))
```
Comment (rcaplier) [inline libs/ui/inputs/src/lib/url-input/url-input.component.ts:21]: Is this needed ? It emits falsy value sometimes ?

File: tailwind.base.css [hunk 2]
```diff
@@ -135,10 +131,26 @@
     @apply flex items-center justify-center px-4 py-1 text-white rounded backdrop-blur;
   }
 
+  /* GENERIC TEXT INPUT (textarea, text field, url input...) */
+  .gn-ui-text-input {
+    --rounded: var(--gn-ui-text-input-rounded, 0.25em);
+    --padding: var(--gn-ui-text-input-padding, 0.6em);
+    @apply appearance-none border-2 p-[--padding] rounded-[--rounded] w-full transition-colors duration-75
```
Comment (rcaplier) [inline tailwind.base.css:138]: That's very nice and is so much nicer that the previous way of doing the base css class string !! I really need to take a deeper look at `@apply`.

**Review by rcaplier — APPROVED** [review]
LGTM ! Very nice job on refactoring the UI components.

**Comment by jahow** [comment]
Thanks for the review!

File: libs/ui/inputs/src/lib/url-input/url-input.component.ts (hunk 1 above)
Comment (rcaplier) [reply libs/ui/inputs/src/lib/url-input/url-input.component.ts:21]: Oh ok, I would have just put a `if (value.trim() === '') return` in the handleChange method.

**Pull request merged**